src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/resource/ComparisonController.kt

```diff
@@ -147,6 +150,58 @@ class ComparisonController(
     return comparisonService.getComparisonPersonByShortReference(comparisonReference, mismatchReference)
   }
 
+  @PostMapping(value = ["{comparisonReference}/mismatch/{mismatchReference}/discrepancy"])
+  @PreAuthorize("hasAnyRole('ROLE_RELEASE_DATE_COMPARER')")
+  @Operation(
+    summary = "Create a comparison person discrepancy record",
+    description = "This endpoint will create a new comparison person discrepancy and return a summary of it",
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "200", description = "New discrepancy created and summary returned"),
+      ApiResponse(responseCode = "401", description = "Unauthorised, requires a valid Oauth2 token"),
+      ApiResponse(responseCode = "403", description = "Forbidden, requires an appropriate role"),
+    ],
+  )
+  fun createComparisonPersonDiscrepancy(
+    @Parameter(required = true, example = "A1B2C3D4", description = "The short reference of the comparison")
+    @PathVariable("comparisonReference")
+    comparisonReference: String,
+    @Parameter(required = true, example = "A1B2C3D4", description = "The short reference of the mismatch")
+    @PathVariable("mismatchReference")
+    mismatchReference: String,
+    @Valid
+    @RequestBody
+    discrepancyInput: CreateComparisonDiscrepancyRequest,
+  ): ComparisonDiscrepancySummary {
+    log.info("Request received to create a new discrepancy -- $discrepancyInput")
+    return comparisonService.createDiscrepancy(comparisonReference, mismatchReference, discrepancyInput)
+  }
+
+  @GetMapping(value = ["{comparisonReference}/mismatch/{mismatchReference}/discrepancy"])
+  @PreAuthorize("hasAnyRole('ROLE_RELEASE_DATE_COMPARER')")
```

Comment:
Should this include ROLE_RELEASE_DATE_MANUAL_COMPARER?

Reply:
as above, I've added to ManualComparisonController.